### DISCUSSION TITLE
dipole model

### DISCUSSION THREAD

**Discussion by chtchelkatchev** [deepmodeling/deepmd-kit#1893] [General]
I have data (forces, virials, energies and some per-atom vectors: "atomic dipoles"). I can easily train a deepmd potential using the dipole model. The question is how to include a vector (let's call it "atomic dipole") in lammps (in-file and data file) so deepmd potential in lammps would understand different configurations of these atomic dipoles and adjust the force field (I do not currently need any dipole-dipole interaction and other complicated long-range things)? Unfortunately I did not see proper answer in examples.

**Comment by wanghan-iapcm** [comment]
By the DPLR model, the atomic dipole model is used to predict the relative distance of the Wannier centroid to a certain atom. Please check examples/water/dplr  and the paper on the DPLR model https://doi.org/10.1063/5.0083669

**Reply by chtchelkatchev** [reply]
Thank You for quick answer. I'll clarify my question then. Individual atoms in my system are characterized by a scalar (in a complex case, a vector) tunable parameters. Can I create a deepmd model depending on the parameters? If yes, how I can tune these individual atomic parameters in lammps? I know about atomparam and how to train a model using it. Unfortunately, it is not clear, how one can tune atomparam in lammps (fparam is clear).

**Reply by wanghan-iapcm** [reply]
By lammps input script one may only set a uniform aparam for all atom. To use different aparam to different atoms you may need some hack of the code. Please see `PairDeepMD::make_ttm_aparam` and `source/lmp/deepmd_fix_ttm.patch`.

To switch-on training of a  model with aparam, one need to set a non zero value of `numb_param` to the fitting net,  see https://docs.deepmodeling.com/projects/deepmd/en/master/train/train-input.html#argument:model/fitting_net[ener]/numb_aparam
A file aparam.npy is required to provide the value of aparam for each atom.

**Reply by chtchelkatchev** [reply]
Thank You for quick answer! 
I successfully trained model from deepmd-kit-213/examples/fparam/train with aparam activated. After that I managed running lammps with different aparam for all atoms modifying only one line in the input script given in examples: "pair_style	deepmd aparam.pb aparam aparam.dat". Different aparam for all atoms were stored in the file aparam.dat that has simple structure (for 4 particles): "0.27 0.23 0.26 0.25". Is it correct run of lammps?
Unfortunately I do not clearly understand what to do with the patch mentioned by You (I patched nothing to run the example). 
Another important question about any limitations of the range of possible values of aparam while training the model. Can I take also negative aparam for some atoms (I really need that)? or aparam with an absolute value more than 1.0?

**Reply by wanghan-iapcm** [reply]
If you want to set the aparam by a file, you have to implement it in the pair style pair_deepmd.cpp. The TTM patch is an example showing how to get aparam from the `fix ttm`.

**Reply by chtchelkatchev** [reply]
Thank You for quick answer! Please, include aparam activation in lammps for future versions of deepmd. We are open for collaboration.
